lib/iris/coords.py

```diff
@@ -1578,6 +1578,28 @@ def bounds(self, bounds):
         self._bounds = bounds
 
 
+class CellMeasures(AuxCoord):
```

Comment:
I've taken a look at the [CF spec docs](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.6/build/cf-conventions.html#cell-measures) for cell measures and had a chat with @marqh about this. Based on these I think that this PR represents a great first step to adding cell measures to Iris but is not quite using the right approach.

Cell measures in the CF spec are dataset variables, meaning that they are in many ways equivalent to the dataset variables that describe dataset coordinates, and we should be treating them as such in Iris. This is not least because the CF spec makes it clear that cell measures can include attributes such as a `standard name` and `units`, and it would be a bit odd to have these attributes hidden away in a coordinate that is part of a cube's attributes dictionary.

I think then that your original plan of making cell measures like `AuxCoord`s in Iris was a good one. However at the same time they are **not** `AuxCoord`s and should not be completely treated like them by Iris &ndash; for example, I would not expect cell measures on an Iris cube to show up in a list of the cube's `AuxCoord`s. Instead it would be good if the cell measures could be accessed via a new cube attribute `cube.cell_measures`. You can also sort of see that we should be tending this way by the fact that in the new load rules you've added you load the cell measures as an `AuxCoord` and then put this coord wholesale into the cube's attributes dictionary.